### DISCUSSION TITLE
What is my Version? :-)

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#150] [Q&A] [answered]
I do  Update-Module -Name PnP.PowerShell
and
Install-Module -Name PnP.PowerShell -RequiredVersion 1.2.0

but get-Module PnP.PowerShell Shows version 0.3.28 

ModuleType Version    Name                                ExportedCommands
----------       -------    ----                                ----------------
Manifest        0.3.28     PnP.PowerShell                      {Add-PnPAlert, Add-PnPApp, Add-

So did it update or not? 
Thank you for a great Module

**Comment by MattK-CPA** [comment, marked as answer]
Hello,
As at this moment, the latest version is version 1.2.4-nightly but that is considered a Prerelease version.
The latest main version is 1.2.0.  
You can always see for yourself: 
`find-Module -Name PnP.PowerShell -AllowPrerelease -AllVersions`

Based upon the information presented, it would seem that you local environment did not update.  It will help you to also observe carefully into, which Directory or Path, the PnP.PowerShell module was installed. 
`get-Module -Name PnP.PowerShell -ListAvailable | select-Object -Property Name, Version, Path`

It may then become a little complicated, because it depends what version of PowerShell you are using eg.  Windows PowerShell 5 or Powershell 7 and what version of the **PowerShellGet** module you are using: 
`get-Module -Name PowerShellGet -ListAvailable`
If you have an appropriate version of PowerShellGet, then you will get access to the AllowPrerelease switch, as in 
`install-Module -Name PnP.PowerShell -AllowPrerelease` 
and then you can use the very very latest nightly build if you want to.

Also be aware of the information in the articles that are available at https://pnp.github.io/powershell/articles/upgrading.html 

Once you do an Install for a module, you generally do not need to do it again (unless for example you uninstall all versions of that module because you want to start clean).  After the initial Install of the module you could just issue update module commands) for example: 
`update-Module -Name PnP.PowerShell`
or
`update-Module -Name PnP.PowerShell -AllowPrerelease`

Hope this information assists you somehow to determine the problem you are experiencing further.

**Reply by Ofer-Gal** [reply]
Turned out that some old 0.3.5 was blocking the new install.
Only deleting the folder made it go away. 
Then` Install-Module Microsoft.Graph -AllowClobber -Force`
Did it and now everything is 1.2.0

Thank you

**Reply by ToddKlindt** [reply]
To keep the old versions from piling up I will quite often run `Uninstall-Module PnP.PowerShell -AllVersions -Force ` in a new admin window before I install a new version.

**Reply by Ofer-Gal** [reply]
Thanks
